### PR TITLE
Admin search: update close button behaviour

### LINKED ISSUE #6862: Admin Page: need to click twice to remove searchform
#### Steps to reproduce the issue

1. Go to Jetpack > Settings.
2. Click on the search icon.
3. Launch a search. Find what you're looking for, make changes.
4. Click on the X icon to remove the searchform.
5. Where are the tabs I'd like to use to navigate to different settings? They're still hidden, I need to click once more on X to fully remove the searchform.

screen recording 2017-04-03 at 12 39 pm

### PR DESCRIPTION
**Pull request by monsieur-z**
Fixes https://github.com/Automattic/jetpack/issues/6862

## Proposed changes:

The behavior of the search bar close button on the Jetpack Settings page is confusing. Clicking it once clears the search input, and clicking it twice collapses the search bar, but the button appearance stays the same. See the linked issue.

This PR adds a button to Clear the search input. In details, it:
- Moves the close button to its own file
- Updates the behavior of the close button to only close the search bar
- Adds a clear button that resets the search input
- Adds a visible focus state to both buttons
- Ensures the search can be used with a keyboard

_Updated search bar_
Screenshot 2024-07-04 at 2 26 05 PM

_Focus states_
Screenshot 2024-07-04 at 2 41 12 PMScreenshot 2024-07-04 at 2 41 14 PM

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

n/a

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

- Spin up a Jetpack site with this branch
- Visit the Jetpack Settings page at `/wp-admin/admin.php?page=jetpack#/settings`
- Test the search
  - The clear button should clear the input
  - The close button should collapse the search bar altogether
- Make sure it works when using a keyboard only

### CODE REVIEW

File: projects/plugins/jetpack/_inc/client/components/search/close-btn.jsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import Gridicon from 'components/gridicon';
+
+const SearchCloseButton = ( { instanceId, closeSearch, closeListener } ) => {
```
Comment (monsieur-z) [inline projects/plugins/jetpack/_inc/client/components/search/close-btn.jsx:3]: Copied from `index.jsx`. No changes were made apart from passing the listeners and state values as props.

File: projects/plugins/jetpack/_inc/client/components/search/index.jsx [hunk 2]
```diff
@@ -13,6 +15,12 @@ import './style.scss';
  */
 const SEARCH_DEBOUNCE_MS = 300;
 
+/**
```
Comment (monsieur-z) [inline projects/plugins/jetpack/_inc/client/components/search/index.jsx:18]: Fix linting issues.

File: projects/plugins/jetpack/_inc/client/components/search/index.jsx [hunk 3]
```diff
@@ -36,7 +44,6 @@ class Search extends React.Component {
 		onSearch: PropTypes.func.isRequired,
 		onSearchChange: PropTypes.func,
 		onSearchOpen: PropTypes.func,
-		onSearchClose: PropTypes.func,
```
Comment (monsieur-z) [inline projects/plugins/jetpack/_inc/client/components/search/index.jsx:39]: A search in the codebase revealed the prop `onSearchClose` was never used.

File: projects/plugins/jetpack/_inc/client/components/search/index.jsx [hunk 4]
```diff
@@ -104,10 +110,6 @@ class Search extends React.Component {
 				: this.props.onSearch;
 		}
 
-		if ( nextProps.isOpen ) {
```
Comment (monsieur-z) [inline projects/plugins/jetpack/_inc/client/components/search/index.jsx:107]: Prevent expanding the search bar again after clicking the close button.

File: projects/plugins/jetpack/_inc/client/components/search/index.jsx [hunk 5]
```diff
@@ -343,6 +345,7 @@ class Search extends React.Component {
 						className={ inputClass }
 						placeholder={ placeholder }
 						role="searchbox"
+						tabIndex={ isOpenUnpinnedOrQueried ? '0' : '-1' }
```
Comment (monsieur-z) [inline projects/plugins/jetpack/_inc/client/components/search/index.jsx:348]: Prevent focusing on the search input when the search bar is collapsed.

**Review by coder-karen — APPROVED** [review]
Tests well, and looks good, thanks for working on this!

**Pull request merged**